### DISCUSSION TITLE
How to create an input (.fst) file to run a simulation?

### DISCUSSION THREAD

**Discussion by tlemoignan** [OpenFAST/openfast#950] [Q&A]
Hi,

I am trying to use OpenFAST for a university project, but so far I am a bit lost with how to go about this. I am using Windows and have downloaded the binaries from https://github.com/OpenFAST/openfast/releases and was hoping to get some help as to where to go from here.

I understand to run a simulation I need to open a command prompt in the directory containing my input file and enter something like:

`>OpenFAST_x64.exe input.fst
`

But I am unsure how to find/create an input file, I'm assuming this file contains all the input parameters needed to run a simulation.

Is this the right approach to be taking if I want to generate some FOWT motion data?

Any help would be greatly appreciated, thank you!
Tom

**Comment by jjonkman** [comment]
Dear @tlemoignan,

Yes, that is all correct.  NREL has supplied a set of full OpenFAST input files in the regression test (r-test) repository--see: https://github.com/OpenFAST/r-test.  Do you already know which FOWT you are trying simulate?

Best regards,

**Reply by tlemoignan** [reply]
Dear @jjonkman ,

Thank you for the quick response! I have since looked through the r-test repository and managed to run a simulation.

I'm looking to compare some simulation data to that of the Equinor Hywind Scotland dataset, the FOWTs used in this wind farm I believe are SWT-6.0-154 Siemens Gamesa spar type turbines.

Ideally I would like to simulate 6 DoF time series motion data for multiple wind/wave/current states, is this likely to be possible?

Thanks again and merry Christmas!
Tom

**Reply by jjonkman** [reply]
Dear @tlemoignan,

The closest model we have in the public domain to that is the NREL 5-MW baseline wind turbine atop the OC3-Hywind spar: https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/5MW_OC3Spar_DLL_WTurb_WavesIrr.  You can certainly simulate the 6 DOF motion response for different wind/wave/current states.  But you'll need to modify this model with actual data for the SWT-6.0-154 turbine and Hywind spar (if you have access to such data) if you want to the results to more closely match measurements from that specific floating wind farm.

Best regards,

**Reply by tlemoignan** [reply]
Dear @jjonkman ,

Since your suggestion I have been playing around with the input files for the spar type NREL 5MW baseline turbine, to try and match the environmental conditions and loading scenario to some of the real world data I have. I have managed to run simulations to output some 6DoF time series data, but when graphing the data it seems there isn't a static equilibrium as there is a mean heave value. I get the feeling this has something to do with the mooring input file, but I was wondering if you might have any other suggestions of how to fix this?

I have attached the heave time series graph over 1000s.

Thanks,
Tom

![image](https://user-images.githubusercontent.com/96588988/152986903-623a7c8b-b002-41e0-828e-bd95edb803e3.png)

**Reply by jjonkman** [reply]
Dear @tlemoignan,

You mention modifing the input files to match environmental conditions and loading scenario, but have you modified the input files to account for a system configuration change (mass, stiffness, geometry, etc.)?

We typically recommend that a floating wind turbine model be set up so that static equilibrium in the vertical direction is achieved through zero heave displacement, i.e., check that the external buoyancy from displaced water balances with the weight of the system (including the weight of the rotor-nacelle assembly, tower and substructure) and vertical mooring system pretension.  The original model of the NREL 5-MW baseline wind turbine atop the OC3-Hywind spar was set up this way; did you change any inputs that would have affected this?

Best regards,

**Reply by tlemoignan** [reply]
Dear @jjonkman,

Thanks again for the help.

To begin with I only modified the inputs relating to the environmental conditions, this produced some nice data which in the frequency domain, while not perfect, still showed some resemblance to the real world data I have.

I am now hoping to modify the input files relating to the system configuration to try and improve the representation of my real world data, and was hoping to ask a few questions on how to go about doing this, you mention maintaining static equilibrium by balancing the buoyancy, system weight and vertical mooring pretension:

1. With regards to the system weight, I only have access to the dry mass of the whole system, so I was intending on making assumptions for the masses of the different parts of the structure, ensuring they total up to this known value. For the tower, I only intend on changing the variables TowerHt and TowerBsHt, keeping NRELOffshrBsline5MW_OC3Hywind_ElastoDyn_Tower.dat unchanged, would I be correct in assuming that the tower mass will just extrapolate from the original value of 249,718kg based on the length of the tower or should I be approaching this differently?

2. Do I need to take into account the mass of the blades separate to HubMass? And if so, considering I only intend on changing TipRad and HubRad, would I be able to apply the same method as I intend on using to calculate the tower mass?

3. Do I need to take into account the mass of the mooring lines?

4. For the mooring system pretension; I have values for the mooring tension, as well as the expected vertical load from the mooring system, is there a way to input these myself, or does OpenFAST calculate them based on the input parameters? 

5. I've set up my MAP file similar to the original, by simplifying the original multi-segment lines to be equivalent homogeneous lines, I understand from https://www.nrel.gov/docs/fy10osti/47535.pdf that this requires the system to be augmented with a yaw spring to achieve the proper overall yaw stiffness, I was wondering how I might calculate this?

6. Finally, by changing the system configuration, I understand this will change the inertia, stiffness and damping of the system, if I'm honest changing all of these values seems daunting, particularly the .hst, .1 and .3 PotFiles, do you have any advice on what absolutely needs to be changed? Also are there any available papers that might help me calculate such values for my system?

Any help would be greatly appreciated!

Kind regards,
Tom

**Reply by jjonkman** [reply]
Dear @tlemoignan,

Here are my answers to your questions:

1. For purely geometric scaling (all spatial dimensions scaling equally), tower mass would scale with the cube of the tower length, but by changing only the tower length without changing the distributed mass per unit length in the tower input file, the tower mass would scale linearly with the tower length. This implies that the diameter and thickness of the tower are not changing with length.  I can't say whether this is a reasonable assumption or not and it doesn't sound like you have the data to confirm.
2. A similar comment to (1) applies to the blade.  It is important to distribute the blade mass within the blade (rather lumping the blade mass into the hub) if you intend to examine structural loads within the blade.
3. In my vertical equilibrium balance, the full system weight should not include the mooring line mass.  The latter would be accounted for in the vertical mooring pretension.
4. MAP++ will calculate the vertical mooring pretension from the mooring system properties provided; you cannot enter the vertical mooring pretension independently from the mooring system properties.
5. In the original OC3-Hywind model, the yaw spring was derived based on knowledge of the yaw-moment relationship for the mooring system that included the mooring delta arrangement.
6. I can't really provided detailed guidance.  Changing the many input parameters is fairly straightforward if you have the properties to be specified.  But it does not sound like you have the properties you need to generate many of the OpenFAST inputs, so, you'll likely need to make many assumptions along the way.  I would not recommend changing the WAMIT output files (_*.hst_, _*.1_, _*.3_) by hand; these should be generated by running WAMIT (or an equivalent potential-flow frequency-domain wave-body interaction solver).  And I'm not aware that detailed properties of the Hywind-Scotland system are available in the public domain.

I hope that helps.

Best regards,

**Reply by tlemoignan** [reply]
Dear @jjonkman,

Thank you.

I just have a few follow-up questions if that's okay, with regards to:

2 - I don't intend on lumping the blade mass into the hub, I was just wondering if you took into account the blade masses when doing the vertical equilibrium balance? I am only really concerned about the 6DoF motions of the FOWT, rather than structural loads.
4 - Is there a way to know the vertical mooring pretension calculated by MAP++?
5 - Is it important then that I derive a value for this for the Hywind Scotland system? I understand that this makes up the yaw term in the additional linear stiffness matrix in HydroDyn.
6 - I'm not sure if I will be able to access WAMIT or equivalent. If this is the case, I guess leaving the WAMIT output files as they are wouldn't work out, would turning off the potential flow model be an option?

Kind regards,
Tom

**Reply by jjonkman** [reply]
Dear @tlemoignan,

Here are my answers to your follow-up questions:

2. Yes, you should account for the blade mass in the system weight.
4. The vertical mooring pretension from the mooring system at the undisplaced position of the platform is calculated by MAP++ and written to the MAP++ summary file.
5. If the yaw spring is not set appropriately--e.g., if the yaw stiffness is too low--the floater yaw motion may become unstable.
6. Given the slenderness of the Hywind spar, modeling the spar via the strip theory solution in HydroDyn is likely a reasonable alternative to the use a hybrid potential-flow plus strip-theory solution that the original OpenFAST model of the OC3-Hywind spar is modeled.

Best regards,
